src/Features/Core/Portable/ExtractMethod/MethodExtractor.CodeGenerator.cs

```diff
@@ -114,6 +116,12 @@ public async Task<GeneratedCode> GenerateAsync(CancellationToken cancellationTok
                 return await CreateGeneratedCodeAsync(result.Status, finalDocument, cancellationToken).ConfigureAwait(false);
             }
 
+            protected virtual Task<SemanticDocument> UpdateMethodAfterGenerationAsync(
```

Comment:
big 👍  on this.

Reply:
The thought process here was that extract method _could_ generate a new return statement. I don't think that's true though, after giving it more thought. If it ever was, we'd want to analyze with the newly generated code as well. This can be front loaded